pkg/provisioner/provisioner.go

```diff
@@ -200,6 +200,9 @@ type Provisioner interface {
 
 	// RemoveBMCEventSubscriptionForNode delete the subscription
 	RemoveBMCEventSubscriptionForNode(subscription metal3api.BMCEventSubscription) (result Result, err error)
+
+	// GetFirmwareComponentsForNode gets all firmware components available from a note
+	GetFirmwareComponentsForNode() (components []metal3api.FirmwareComponentStatus, err error)
```

Comment:
We don't use the ForNode suffix consistently here. I'd stay close to GetFirmwareSettings and drop it.